crates/nu-command/src/network/url/parse.rs

```diff
@@ -0,0 +1,184 @@
+use super::url;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, PipelineData, ShellError, Signature, Span, SyntaxShape, Value,
+};
+
+use url::Url;
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "url parse"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("url parse")
+            .rest(
+                "rest",
+                SyntaxShape::CellPath,
+                "optionally operate by cell path",
+            )
+            .category(Category::Network)
+    }
+
+    fn usage(&self) -> &str {
+        "Parses a url"
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["url", "parse"]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        _: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        parse(input.into_value(call.head), engine_state)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Parses a url",
+            example: "echo 'http://www.example.com/foo/bar?p=section&s=&f[name]=vldc' | url parse",
```

Comment:
Although several commands still do it in their examples, `echo` isn't necessary any longer and for cases like this IMO we should omit the `echo`: a data structure can just be supplied as input to a pipeline.